service/src/main/java/bio/terra/user/app/controller/ProfileApiController.java

```diff
@@ -19,32 +22,47 @@ public class ProfileApiController implements ProfileApi {
   private final HttpServletRequest servletRequest;
   private final ProfileService profileService;
   private final ProfileSamUserFactory userFactory;
+  private final SamService samService;
 
   @Autowired
   public ProfileApiController(
       HttpServletRequest servletRequest,
       ProfileService profileService,
-      ProfileSamUserFactory userFactory) {
+      ProfileSamUserFactory userFactory,
+      SamService samService) {
     this.servletRequest = servletRequest;
     this.profileService = profileService;
     this.userFactory = userFactory;
+    this.samService = samService;
   }
 
   @Override
-  public ResponseEntity<AnyObject> setUserProfile(AnyObject body, String path) {
-    profileService.setProperty(getUser(), parsePath(path), body.getValue());
+  public ResponseEntity<AnyObject> setUserProfile(
```

Comment:
I still think the code should be more clear that it's checking auth in the beginning. If you look at WSM WorkspaceApiController.java, every method either checks auth in the beginning (validateWorkspaceAndAction()), or has a comment explaining why auth is not checked. This way, we know where to look to confirm that we're checking auth; we look in controller, not service or dao. I think it's ok to make to make an extra SAM call for this.

Create SamService.checkAdminAuthz to be more clear.

```
if (!StringUtils.isEmpty(userEmail)) {
      AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
    SamRethrow.onInterrupted(
        () -> getSamService().checkAdminAuthz(userRequest),
        "check whether the user has admin access");
}
```

Or put above code into ProfileApiController.maybeCheckRequesterIsSamAdmin() and do:

```
maybeCheckRequesterIsSamAdmin();
```

Or create ProfileApiController.checkRequesterIsSamAdmin() and:

```
if (!StringUtils.isEmpty(userEmail)) {
  checkRequesterIsSamAdmin();
}
```

Reply:
Thanks! I added the `checkRequesterIsSamAdmin` check.